src/test/ui/function-pointer/unsized-ret.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0277]: the size for values of type `str` cannot be known at compilation time
+  --> $DIR/unsized-ret.rs:9:5
+   |
+LL |     foo::<fn() -> str, _>(None, ());
+   |     ^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
+   |
+   = help: within `fn() -> str`, the trait `Sized` is not implemented for `str`
+   = note: required because it appears within the type `fn() -> str`
+note: required by a bound in `foo`
+  --> $DIR/unsized-ret.rs:4:11
+   |
+LL | fn foo<F: Fn<T>, T>(f: Option<F>, t: T) {
+   |           ^^^^^ required by this bound in `foo`
```

Comment:
The implicit `Sized` requirements should have their own obligation cause so that we could mention *why* it was required (as the code doesn't mention it, this would be mystifying the first time you encounter it).